src/cmd/run.js

```diff
@@ -257,14 +262,21 @@ export default async function run(
       }
 
       log.info('The extension will reload if any source file changes');
+      const createWatcher = (...args) => {
+        const fileFilter = new FileFilter({
+          sourceDir, artifactsDir, ignoreFiles,
+        });
+        args[0].shouldWatchFile = (file) => fileFilter.wantFile(file);
+        return defaultWatcherCreator(...args);
+      };
```

Comment:
You're right. I should make `reloadStrategy` accept `ignoreFiles` and build `shouldWatchFile` inside `reloadStrategy`.

Also maybe we should remove `FileFilter` from the watcher so it doesn't depend on `FileFilter` anymore. Make `shouldWatchFile` default to `() => true`.